### ISSUE TITLE
Iris example is not running

### ISSUE THREAD

**Issue by homonecloco** [JedS6391/LGP-examples#1]
Hi, 

I'm trying to figure out how to run examples more complex than the simple without any operations, I found this repository, andI'd been able to run all the tests except the Iris, example, that gets this Exception

```
N109338:libs ramirezr$ kotlin -cp LGP-examples-1.0.jar:. nz.co.jedsimson.lgp.examples.kotlin.Iris
Exception in thread "main" java.util.concurrent.ExecutionException: kotlin.NotImplementedError: An operation is not implemented.
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at nz.co.jedsimson.lgp.core.evolution.training.DistributedTrainer.train(DistributedTrainer.kt:178)
	at nz.co.jedsimson.lgp.examples.kotlin.IrisProblem.solve(Iris.kt:188)
	at nz.co.jedsimson.lgp.examples.kotlin.Iris$Main.main(Iris.kt:209)
	at nz.co.jedsimson.lgp.examples.kotlin.Iris.main(Iris.kt)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.jetbrains.kotlin.runner.AbstractRunner.run(runners.kt:61)
	at org.jetbrains.kotlin.runner.Main.run(Main.kt:110)
	at org.jetbrains.kotlin.runner.Main.main(Main.kt:120)
Caused by: kotlin.NotImplementedError: An operation is not implemented.
	at nz.co.jedsimson.lgp.lib.generators.EffectiveProgramInstructionGenerator.generateInstruction(EffectiveProgramGenerator.kt:36)
	at nz.co.jedsimson.lgp.core.program.instructions.InstructionGenerator$next$1.invokeSuspend(InstructionGenerator.kt:38)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:32)
	at kotlin.sequences.SequenceBuilderIterator.hasNext(SequenceBuilder.kt:137)
	at nz.co.jedsimson.lgp.lib.generators.EffectiveProgramGenerator.generateProgram(EffectiveProgramGenerator.kt:211)
	at nz.co.jedsimson.lgp.core.program.ProgramGenerator$next$1.invokeSuspend(ProgramGenerator.kt:33)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:32)
	at kotlin.sequences.SequenceBuilderIterator.hasNext(SequenceBuilder.kt:137)
	at kotlin.sequences.TakeSequence$iterator$1.hasNext(Sequences.kt:376)
	at kotlin.sequences.SequencesKt___SequencesKt.toCollection(_Sequences.kt:691)
	at kotlin.sequences.SequencesKt___SequencesKt.toMutableList(_Sequences.kt:721)
	at nz.co.jedsimson.lgp.core.evolution.model.Models$SteadyState.initialise(Models.kt:92)
	at nz.co.jedsimson.lgp.core.evolution.model.Models$SteadyState.train(Models.kt:100)
	at nz.co.jedsimson.lgp.core.evolution.training.DistributedTrainer$ModelTrainerTask.call(DistributedTrainer.kt:145)
	at nz.co.jedsimson.lgp.core.evolution.training.DistributedTrainer$ModelTrainerTask.call(DistributedTrainer.kt:131)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

```

**Comment by JedS6391** [comment]
Hi @homonecloco, I have identified the issue that is causing this (it is due to a use of branching operations for that particular problem). 

I am aiming to release some new versions for the LGP-core and LGP-lib packages to address this (among other issues), and will update the examples accordingly. I will provide an update here when the issue has been resolved.

**Comment by JedS6391** [comment]
Hi @homonecloco, this issue should be resolved in the latest commit (073ea679cf81ee2b02b80e02f59e059e9381a66e).

Please let me know if you are still encountering the issue and I'll look into it further.

**Comment by homonecloco** [comment]
I tested and it works.
